Corona-Warn-App/src/main/java/de/rki/coronawarnapp/deadman/DeadmanNotificationScheduler.kt

```diff
@@ -4,16 +4,55 @@ import androidx.work.ExistingPeriodicWorkPolicy
 import androidx.work.ExistingWorkPolicy
 import androidx.work.WorkManager
 import dagger.Reusable
+import de.rki.coronawarnapp.coronatest.CoronaTestRepository
+import de.rki.coronawarnapp.nearby.ENFClient
+import de.rki.coronawarnapp.storage.OnboardingSettings
+import de.rki.coronawarnapp.util.coroutine.AppScope
+import de.rki.coronawarnapp.util.flow.combine
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 import timber.log.Timber
 import javax.inject.Inject
 
 @Reusable
 class DeadmanNotificationScheduler @Inject constructor(
+    @AppScope val appScope: CoroutineScope,
     val timeCalculation: DeadmanNotificationTimeCalculation,
     val workManager: WorkManager,
-    val workBuilder: DeadmanNotificationWorkBuilder
+    val workBuilder: DeadmanNotificationWorkBuilder,
+    val onboardingSettings: OnboardingSettings,
+    val enfClient: ENFClient,
+    val coronaTestRepository: CoronaTestRepository
 ) {
 
+    fun setup() {
+        Timber.i("setup() DeadmanNotificationScheduler")
+
+        combine(
+            onboardingSettings.isOnboardedFlow,
+            coronaTestRepository.coronaTests,
+            enfClient.isTracingEnabled
+        ) { isOnboarded, coronaTests, isTracingEnabled ->
+            val noPositiveTestRegistered = coronaTests.none { it.isSubmissionAllowed }
```

Comment:
Should we use `isPositive`?  Because isSubmissionAllowed will be false again after submission.

```kotlin
    override val isSubmissionAllowed: Boolean
        get() = isPositive && !isSubmitted
```

Reply:
Good point. Thanks! Changed it to use `isPositive` now.